src/parser/Parser.cpp

```diff
@@ -3,98 +3,105 @@
+
+    if (m_currenttoken.type == TokenType::Print) {
+        eattoken(TokenType::Print);
         std::unique_ptr<AST::Expr> expr = parseExpr();
-        currTok = m_lexer.getToken();
-        if (currTok.type != TokenType::EOL) {
-            // TODO syntax error, synchronize
-        }
         return std::unique_ptr<AST::PrintStmt>(new AST::PrintStmt(std::move(expr)));
     }
-    if (currTok.type == TokenType::Err) {
+
+    if (m_currenttoken.type == TokenType::Err) {
+        eattoken(TokenType::Err);
         std::unique_ptr<AST::Expr> expr = parseExpr();
-        currTok = m_lexer.getToken();
-        if (currTok.type != TokenType::EOL) {
-            // TODO syntax error, synchronize
-        }
         return std::unique_ptr<AST::ErrStmt>(new AST::ErrStmt(std::move(expr)));
     }
-    if (currTok.type == TokenType::Scan) {
-        // TODO
-        Token ident = m_lexer.getToken();
-        if (ident.type != TokenType::Identifier) {
-            // TODO syntax error, synchronize
-        }
-        currTok = m_lexer.getToken();
-        if (currTok.type != TokenType::EOL) {
-            // TODO syntax error, synchronize
-        }
+
+    if (m_currenttoken.type == TokenType::Scan) {
+        eattoken(TokenType::Scan);
+        Token id = eattoken(TokenType::Identifier);
+        return std::unique_ptr<AST::ScanStmt>(new AST::ScanStmt(std::unique_ptr<AST::Identifier>(new AST::Identifier(id.literal))));
     }
-    if (currTok.type == TokenType::Def) {
+
+    if (m_currenttoken.type == TokenType::Def) {
         // TODO
     }
-    if (currTok.type == TokenType::Return) {
+
+    if (m_currenttoken.type == TokenType::Return) {
         std::unique_ptr<AST::Expr> expr = parseExpr();
-        currTok = m_lexer.getToken();
-        if (currTok.type != TokenType::EOL) {
+        m_currenttoken = m_lexer.getToken();
+        if (m_currenttoken.type != TokenType::EOL) {
             // TODO syntax error, synchronize
         }
         return std::unique_ptr<AST::ReturnStmt>(new AST::ReturnStmt(std::move(expr)));
     }
+
     // assignment or expr statement
     std::unique_ptr<AST::Expr> expr = parseExpr();
-    currTok = m_lexer.getToken();
-    if (currTok.type == TokenType::EOL) {
-        return expr;
-    }
-    if (currTok.type == TokenType::Equal) {
-        std::unique_ptr<AST::Expr> rhs = parseExpr();
-        return std::unique_ptr<AST::AssignStmt>(new AST::AssignStmt(std::move(expr), std::move(rhs)));
+    if (m_currenttoken.type == TokenType::Equal) {
+        eattoken(TokenType::Equal);
+        std::unique_ptr<AST::Expr> rval = parseExpr();
+        // TODO: add checks that lval is valid.
+        return std::unique_ptr<AST::AssignStmt>(new AST::AssignStmt(std::move(expr), std::move(rval)));
+    } else {
+        return std::unique_ptr<AST::ExprStmt>(new AST::ExprStmt(std::move(expr)));
     }
-    // TODO error handling
-    return nullptr;
+
+    // this is how I do error handling lol.
+    exit(1);
```

Comment:
I assume we'll change this later on?

For syntax errors I do like the panic mode/synchronization stuff done in Crafting Interpreters: http://www.craftinginterpreters.com/parsing-expressions.html#syntax-errors
(though admittedly it's not like I know about anything else).